tests/ut/py/test_elf_parser.py

```
@@ -0,0 +1,296 @@
+        buf.extend(s.encode("ascii") + b"\x00")
+    return bytes(buf), offsets
+
+
+class TestSingleTextNoRelocations:
+    """Kernels whose call graph fully folds into ``.text`` (no
+    ``.text._Z*``, no ``.rela.text``) must continue to load with
+    byte-identical output — this is the regression guard for every
+    kernel that currently works."""
+
+    def test_returns_text_bytes_verbatim(self):
+        text_bytes = b"\xd6\x5f\x03\xc0" * 4  # RET × 4; arbitrary payload, never executed
```

Ruff (0.15.14)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)